hooks/shotgun_get_published_file_for_viewer.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2019 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Hook executed to get a PublishedFile from a Version
+or a PublishedFile (legacy TankPublishedFile supported).
+
+This implementation returns the first published file it finds
+matching extensions in the order of viewer extensions.
+If none are found, raise a TankError.
+"""
+
+import sgtk
+from sgtk import TankError
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class GetPublishedFileForViewer(HookBaseClass):
```

Comment:
Should the class have a different name?

Reply:
Like what?

Reply:
one is called `GetPublishedFile` and the other `GetPublishedFileForViewer`. I don't get it.

Reply:
Fine if it is the case, but this is what I'm seeing
![Screen Shot 2019-09-11 at 16 25 52](https://user-images.githubusercontent.com/39291844/64705887-fa4a0b80-d4b0-11e9-9449-87084c8873de.png)